src/pykeen/nn/modules.py

```diff
@@ -326,6 +328,52 @@ def reset_parameters(self):
                 mod.reset_parameters()
 
 
+class LiteralInteraction(
+    Interaction,
+    Generic[HeadRepresentation, RelationRepresentation, TailRepresentation],
+):
+    """The interaction function shared by literal-containing interactions."""
+
+    def __init__(
+        self,
+        base: HintOrType[Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]],
+        combination: Combination,
+        base_kwargs: Optional[Mapping[str, Any]] = None,
+    ):
+        # TODO documentation for this
+        super().__init__()
+        self.base = interaction_resolver.make(base, base_kwargs)
+        self.combination = combination
+        self.entity_shape = tuple(self.base.entity_shape) + ("e",)
```

Comment:
Do we need to make sure that `"e"` is unused?

Reply:
yes, there needs to be an extra dimension floating around here that gets added in by the literal model

Reply:
Added more docs for this in c7b9bc19